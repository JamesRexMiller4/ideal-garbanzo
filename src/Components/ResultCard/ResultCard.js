import React, { useState } from 'react';
import './ResultCard.scss';

const ResultCard = ({result}) => {
  const [ active, setActive ] = useState(false)
  console.log(result.genre)

  const generateGenreTags = (genres) => {
    let splitUpGenres = genres.split(',');
    return splitUpGenres.map(genre => <button>{genre}</button>)
  }


  return ( 
    <article>
      <div>
        <h3>{result.name}</h3>
        <div>
          <h3>{result.city}</h3>
          <h3>{result.state}</h3>
        </div>
        <h3>{result.phone}</h3>
      </div>
      <h3>{result.genre}</h3>
      {generateGenreTags(result.genre)}
      { active && (
        <div>
          <h4>Additional Info</h4>
        </div>
        )
      }
    </article>
  );
}

export default ResultCard;