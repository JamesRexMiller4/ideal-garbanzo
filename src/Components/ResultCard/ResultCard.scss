@import '../../scss/variables.scss';

article {
  width: 90%;
  border: 3px solid;
  border-radius: .2rem;
  background-color: rgba(248, 248, 248, 0.964);
  padding: 3rem;
  margin: .5rem;
  box-shadow: 0px 8px 8px -4px black;

  &:hover {
    cursor: pointer;
    background-color: $background-blue-gray;
    button {
      background-color: darken($highlight-blue, 10);
      color: #ffffff;
    }
  }

  .result-card-section {
    display: flex;
    justify-content: space-between;
  }

  h3 {
    font-family: 'Open Sans', sans-serif;
    font-size: 1.5rem;
    font-weight: bold;
  }

  h4 {
    font-family: 'Open Sans', sans-serif;
    font-weight: normal;
  }
}

.genre-tag-btn {
  background-color: $background-blue-gray;
  color: $base-blue;
  border: 2px solid;
}

.additional-info-div {
  padding-top: 2rem;
  width: 100%;

  h4 {
    margin-top: 1rem;
    font-weight: bold;
  }

  h5 {
    padding-top: 1rem;
  }

  button {
    margin-top: 1rem;
  }

  section {
    display: flex;
    align-items: space-around;
    justify-content: space-around;
  }
}

.tags-container-div {
  width: 50%;
}