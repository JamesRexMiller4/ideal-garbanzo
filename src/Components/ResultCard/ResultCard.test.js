import React from 'react';
import {shallow} from 'enzyme';
import ResultCard from './ResultCard';
import mockData from '../../data/mockData';
 
describe('ResultCard', () => {
  let wrapper;
 
  beforeEach(() => {
    wrapper = shallow(<ResultCard result={mockData[0]}/>)
  });
 
  it('should render the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render differently when ResultCard is clicked', () => {
    wrapper.simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});