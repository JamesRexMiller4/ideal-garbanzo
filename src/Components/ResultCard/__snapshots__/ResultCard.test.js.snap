// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultCard should render differently when ResultCard is clicked 1`] = `
<article
  onClick={[Function]}
>
  <section
    className="result-card-section"
  >
    <div>
      <h3
        className="result-card-h3"
      >
        Old Hickory Steakhouse
      </h3>
      <h4>
        Oxon Hill, MD
      </h4>
      <h3 />
    </div>
    <div>
      <button
        className="genre-tag-btn"
        key="679199950804.1945"
      >
        Steak
      </button>
      <button
        className="genre-tag-btn"
        key="618223098624.8542"
      >
        American
      </button>
      <button
        className="genre-tag-btn"
        key="197330976093.94327"
      >
        Contemporary
      </button>
      <button
        className="genre-tag-btn"
        key="897193641942.3282"
      >
        Seafood
      </button>
      <button
        className="genre-tag-btn"
        key="47806346992.00138"
      >
        Cafe
      </button>
    </div>
  </section>
  <div
    className="additional-info-div"
  >
    <h3>
      Additional Info
    </h3>
    <section>
      <div>
        <h4>
          Address:
        </h4>
        <h5>
          201 Waterfront St
        </h5>
        <h5>
          Oxon Hill, MD
        </h5>
        <h5>
          20745
        </h5>
      </div>
      <div>
        <h4>
          Phone:
        </h4>
        <h5>
          (301) 965-4000
        </h5>
      </div>
      <div
        className="hours-div"
      >
        <h4>
          Hours of Operation:
        </h4>
        <h5>
          Open Daily 5:30 PM-10:00 PM
        </h5>
      </div>
      <div
        className="tags-container-div"
      >
        <h4>
          Tags:
        </h4>
        <button
          className="genre-tag-btn"
          key="1466322536778.6714"
        >
          Social
        </button>
        <button
          className="genre-tag-btn"
          key="195548699037.32477"
        >
          Food and Dining
        </button>
        <button
          className="genre-tag-btn"
          key="1489214701997.1846"
        >
          Restaurants
        </button>
        <button
          className="genre-tag-btn"
          key="1141115244979.5771"
        >
          Steakhouses
        </button>
      </div>
    </section>
  </div>
</article>
`;

exports[`ResultCard should render the snapshot 1`] = `
<article
  onClick={[Function]}
>
  <section
    className="result-card-section"
  >
    <div>
      <h3
        className="result-card-h3"
      >
        Old Hickory Steakhouse
      </h3>
      <h4>
        Oxon Hill, MD
      </h4>
      <h3 />
    </div>
    <div>
      <button
        className="genre-tag-btn"
        key="434503411844.36865"
      >
        Steak
      </button>
      <button
        className="genre-tag-btn"
        key="720015119374.6793"
      >
        American
      </button>
      <button
        className="genre-tag-btn"
        key="1112164907789.0022"
      >
        Contemporary
      </button>
      <button
        className="genre-tag-btn"
        key="9178867111.464867"
      >
        Seafood
      </button>
      <button
        className="genre-tag-btn"
        key="1125924972531.1506"
      >
        Cafe
      </button>
    </div>
  </section>
</article>
`;
